### ISSUE TITLE
omniscient.d.ts

### ISSUE THREAD

**Issue by polkovnikov-ph** [omniscientjs/omniscient#128]
Is there any support for TypeScript planned?

**Comment by mikaelbr** [comment]
Not officially planed. Did think about doing it earlier, but haven't gotten around to it yet. But AFAIK, anyone could create a type definition and add it to e.g. http://definitelytyped.org/?

**Comment by polkovnikov-ph** [comment]
@mikaelbr Yeah, I'm already up to this painful task. The problem is that there's also no `immstruct.d.ts`, and `immutable.d.ts` exists, but is unavailable from `DefinitelyTyped`.

**Comment by polkovnikov-ph** [comment]
Some preliminary code is [here](https://github.com/polkovnikov-ph/DefinitelyTyped/commits/master), yet `react.d.ts` turned up to be buggy to the backbone, so I think there's no easy way to use anything near React in TypeScript currently. I hesitate fixing bugs in `react.d.ts` myself until I get sure Facebook [is interested in any PRs](https://github.com/facebook/immutable-js/pull/765), because otherwise it's going to be yet another incompatible `react.d.ts`.

**Comment by mikaelbr** [comment]
Ah. Good work and thanks for your efforts! Sad to hear that the type descriptions aren't up to par in React or Immutable.js, but hope to see it improve through your work. Just let me know when you feel it makes sense to include or reference it from this repo.

**Comment by leonyu** [comment]
Saw this issue kind of late, I did have a version floating around here: 
https://github.com/leonyu/DefinitelyTyped/tree/immstruct/immstruct
Although my version used generics (with "abstract cursor implementation"), which is less intuitive to work with, but didn't require reference to `cursor.d.ts`. I went down that route because I was hoping to be able to do it w/o the actual having a reference to `cursor.d.ts` definition, but user experience of that is a lot worse, so I didn't try to get it merged.) 

@polkovnikov-ph I might port over the tests I have into your branch. 

I have been working through the Facebook licensing problem here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/6623. I have been on vacation for the past few weeks. See that the original contributor of that PR is no longer active, I might make a new PR and get it merged.

**Comment by polkovnikov-ph** [comment]
@leonyu Great!
